pallets/ah-migrator/src/vesting.rs

```diff
@@ -86,3 +89,58 @@ pub mod alias {
 		V1,
 	}
 }
+
+#[cfg(feature = "std")]
+impl<T: Config> crate::types::AhMigrationCheck for VestingMigrator<T> {
+	type RcPrePayload =
+		Vec<(Vec<u8>, Vec<BalanceOf<T>>, Vec<GenericVestingInfo<BlockNumberFor<T>, BalanceOf<T>>>)>;
+
+	type AhPrePayload = ();
+
+	fn pre_check(_: Self::RcPrePayload) -> Self::AhPrePayload {
+		let vesting_schedules: Vec<_> = pallet_vesting::Vesting::<T>::iter().collect();
+		assert!(
+			vesting_schedules.is_empty(),
+			"Vesting schedules should be empty before migration starts"
+		);
+	}
+
+	fn post_check(rc_pre_payload: Self::RcPrePayload, _: Self::AhPrePayload) {
+		use std::collections::BTreeMap;
+
+		let rc_pre: BTreeMap<
```

Comment:
I think you can also directly put the BTreeMap as `RcPrePayload` instead of the vector?